logger: added log level parameter

ryanskidmore opened the pull request:
Added a parameter to the `DefaultLogger` to specify log level (in reference to #4)

Defaults to the `error` log level

z4kn4fein commented:
Hey, thanks for the PR! 
Everything looks good, however, we'd like to set the default log level to warning instead of error, what do you think about that?

You can leave the sample unchanged now to let the CI check pass, we'll update it later after the release.

ryanskidmore commented:
Hey @z4kn4fein. Thanks for the feedback! I've reverted the sample changes and changed the default log level to `Warn`.

Please let me know if there are any other issues with this :)

z4kn4fein commented:
Thanks!
